File: src/main/java/no/nav/tjenestepensjon/simulering/domain/Dateable.java
package no.nav.tjenestepensjon.simulering.domain;

import java.util.Date;

public interface Dateable {
    Date getDatoFom();

    static int sortAscendingByFomDato(Dateable o1, Dateable o2) {
        if (o1.getDatoFom().before(o2.getDatoFom())) {
            return -1;
        } else if (o1.getDatoFom().after(o2.getDatoFom())) {
            return 1;
        } else {
            return 0;
        }
    }
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/domain/Dateable.kt
package no.nav.tjenestepensjon.simulering.domain

import java.time.LocalDate

interface Dateable: Comparable<Dateable> {
    var datoFom: LocalDate
    override operator fun compareTo(other: Dateable) = datoFom.compareTo(other.datoFom)
}